routing/fseg_index.hpp

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include "routing/joint.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/unordered_map.hpp"
+#include "std/vector.hpp"
+
+namespace routing
+{
+class RoadJointIds final
+{
+public:
+  JointId GetJointId(uint32_t segId) const
+  {
+    if (segId < m_jointIds.size())
+      return m_jointIds[segId];
+
+    return kInvalidJointId;
+  }
+
+  void AddJoint(uint32_t segId, JointId jointId)
+  {
+    if (segId >= m_jointIds.size())
+      m_jointIds.insert(m_jointIds.end(), segId + 1 - m_jointIds.size(), kInvalidJointId);
```

Comment:
Statistics for Moscow: 645051 normal joint ids, 416526 dummies.
Dummies are more efficient then map or sorted <segId, jointId> pairs.